src/nvim/api/buffer.c

```diff
@@ -757,6 +756,106 @@ void nvim_buf_set_text(uint64_t channel_id, Buffer buffer, Integer start_row, In
   try_end(err);
 }
 
+/// Gets a range from the buffer.
+///
+/// This differs from |nvim_buf_get_lines()| in that it allows retrieving only
+/// portions of a line.
+///
+/// Indexing is zero-based. Column indices are end-exclusive.
+///
+/// Prefer |nvim_buf_get_lines()| when retrieving entire lines.
+///
+/// @param channel_id
+/// @param buffer           Buffer handle, or 0 for current buffer
+/// @param start_row        First line index
+/// @param start_column     First column
+/// @param end_row          Last line index
+/// @param end_column       Last column
+/// @param[out] err         Error details, if any
+ArrayOf(String) nvim_buf_get_text(uint64_t channel_id, Buffer buffer,
+                                  Integer start_row, Integer start_col,
+                                  Integer end_row, Integer end_col, Error *err)
+  FUNC_API_SINCE(9)
+{
+  Array rv = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (!buf || buf->b_ml.ml_mfp == NULL) {
```

Comment:
Not 100% sure but I believe from playing with marks that `buf->b_ml.ml_mfp == NULL` happens on unloaded buffers, in any case I think we might want some explicit error message here. Perhaps even loading the buffer?